File: CS/GridReportPrint/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace GridReportPrint {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }
        public XtraReport1(int CatID):this() {
            new NwindDataSetTableAdapters.ProductsTableAdapter().Fill(this.nwindDataSet1.Products, CatID);
        }

    }
}

File: VB/GridReportPrint/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace GridReportPrint
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Sub New(ByVal CatID As Integer)
			Me.New()
			CType(New NwindDataSetTableAdapters.ProductsTableAdapter(), NwindDataSetTableAdapters.ProductsTableAdapter).Fill(Me.nwindDataSet1.Products, CatID)
		End Sub

	End Class
End Namespace
